osu.Game/Overlays/News/Sidebar/MonthPanel.cs

```diff
@@ -0,0 +1,152 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics;
+using osu.Game.Online.API.Requests.Responses;
+using osu.Game.Graphics.Containers;
+using osuTK;
+using osu.Game.Graphics.Sprites;
+using osu.Game.Graphics;
+using System.Linq;
+using System.Collections.Generic;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics.Sprites;
+
+namespace osu.Game.Overlays.News.Sidebar
+{
+    public class MonthPanel : CompositeDrawable
+    {
+        public readonly BindableBool IsOpen = new BindableBool();
+
+        private readonly FillFlowContainer postsFlow;
+
+        public MonthPanel(List<APINewsPost> posts)
+        {
+            Width = 160;
+            AutoSizeDuration = 250;
+            AutoSizeEasing = Easing.OutQuint;
+            InternalChild = new FillFlowContainer
+            {
+                RelativeSizeAxes = Axes.X,
+                AutoSizeAxes = Axes.Y,
+                Direction = FillDirection.Vertical,
+                Spacing = new Vector2(0, 5),
+                Children = new Drawable[]
+                {
+                    new DropdownButton(posts[0].PublishedAt)
```

Comment:
Not sure we need assert here since I'm pretty sure logic in sidebar will handle everything just right.

Reply:
Well, if you have a year and month integers for the month dropdown, and a set of posts that have published dates, it is not exactly known from the context of `MonthDropdown` alone whether all these posts are within the given month and year, and if one post is not within them, then that's basically wrong and `MonthDropdown` has to throw an exception that something is not right with the given data, that's basically one case of what asserts are for in general I believe.

Reply:
If posts are not within a given year - that will be a web-side issue already then?

Reply:
In case you missed it, I'm suggesting this assertion when adding separate `year` and `month` integers rather than using the date of the first given post, you can't possibly do any assertion currently.

Reply:
So it will be something like this: MonthPanel(month, year, posts) and checking if post dates are matching these ints?
If that's it - I have no any problems with that

Reply:
And use it instead of `posts[0].PublishedAt` in the highlighted line above, which is the main purpose of these integers, yes. 

That's my suggestion from the beginning of this conversation, and the assertion is just for safety measures when the main suggestion itself is applied.

Reply:
Gotcha

Reply:
Done in https://github.com/ppy/osu/pull/12736/commits/f4801c08ff9693c8e97a5252b96a38fe6f288760 commit